### PR TITLE
Added documentation for new functionality to add Dated SNAPSHOTS

### PR DESCRIPTION
**Pull request by axsaucedo**

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **axsaucedo**

If they are not already assigned, you can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Apr  7 12:29:19 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1672/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1672 --build=2

**Comment by seldondev** [comment]
Tue Apr  7 12:29:23 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1672/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1672 --build=1

**Comment by seldondev** [comment]
Tue Apr  7 13:00:26 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1672/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1672 --build=3

**Comment by seldondev** [comment]
Tue Apr  7 13:00:41 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1672/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1672 --build=4

File: doc/source/reference/upgrading.md [hunk 1]
```diff
@@ -24,8 +24,19 @@ The new service orchestrator comes with several advantages including ability to
...
+
+### Dated SNAPSHOTS
+
+Whenever a new PR was merged to master, we have set up our CI to build a "SNAPSHOT" version, which would contain the Docker images for that specific development / master-branch code. 
+
+Previously, we always had the SNAPSHOT tag being overriden with the latest. This didn't allow us to know what version someone may be trying out when using master, so we wanted to introduce a way to actually get unique tags for every image that gets landed into master.
+
+Now every time that a PR is landed to master, a new "dated" SNAPSHOT version is created, which pushes images with the tag `"<next-version>-SNAPSHOT_<timestamp>"`. A new branch is also created with the name `"v<next-version>-SNAPSHOT_<timestamp>"`, which contains the respective helm charts, and allows for the specific version (as outlined by the version in `version.txt`) to be installed.
 
+You can follow the instructions in the installation page to install the snapshot version.
```
Comment (adriangonz) [inline doc/source/reference/upgrading.md:39]: Should we add a link here to the installation page?

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1672#issuecomment-610868698" title="Approved">axsaucedo</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by axsaucedo** [comment]
@adriangonz I just saw your comment, good point, I'll add it - also yes, in master you will always have the latest pointing to <next-version>-SNAPSHOT without date

**Comment by seldondev** [comment]
Wed Apr  8 10:07:30 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1672/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1672 --build=5

**Comment by seldondev** [comment]
Wed Apr  8 10:07:31 UTC 2020
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1672/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1672 --build=6

**Pull request merged**